File: algorithm/src/BubbleSort.java
public class BubbleSort {
    public void bubbleSort(int[]a){
        int length=a.length;
        int temp;
        for (int i = 0; i < a.length; i++) {
            for (int j = 0; j < a.length-i-1; j++) {
                if(a[j]>a[j+1]){
                    temp=a[j];
                    a[j]=a[j+1];
                    a[j+1]=temp;
                }
            }
        }
    }
}

File: algorithm/sort go/bubblesort.go
package sort_go

func BubbleSort(num []int){
	n :=len(num)
	if n<=1 {
		return
	}
	for i:=0;i<n;i++ {
		for j:=i+1;j<n;j++ {
			if (num)[i]>(num)[j]{
				(num)[i],(num)[j]=(num)[j],(num)[i]
			}
		}
	}
}